index/scorch/segment/zap/new.go

```diff
@@ -35,12 +36,11 @@ func AnalysisResultsToSegmentBase(results []*index.AnalysisResult,
 	chunkFactor uint32) (*SegmentBase, error) {
 	var br bytes.Buffer
 
-	s := interim{
-		results:     results,
-		chunkFactor: chunkFactor,
-		w:           NewCountHashWriter(&br),
-		FieldsMap:   map[string]uint16{},
-	}
+	s := interimPool.Get().(*interim)
```

Comment:
One pattern I've seen is to create a wrapper around the sync pool.

1.  The wrapper is strongly typed, and the actual pool isn't exported, so you reduce the liklihood that someone sticks the wrong thing in the pool.

2.  All type assertions happen in the wrapper, so the rest of the code is a little cleaner to read.

3.  The cleanse() method is internal to all items returned to the pool, so you don't have to worry about someone forgetting to cleanse.

Reply:
The counter-argument is that this is all internal anyway.

I'll let you decide if it helps readabilty/usabilty, or if it just creates more cruft.

Reply:
Yah, I think a wrapper would make lots of sense if there were more places where this was used, and right now favor the direct maneuver due to the "this is all internal" current situation.